### DISCUSSION TITLE
Moving hosts - old version to new version

### DISCUSSION THREAD

**Discussion by rthally** [jasonacox/Powerwall-Dashboard#329] [Q&A]
Greetings!

Moving from an intel mac to Apple silicone Mac.  I have a quite old, "un-upgradable" version running on the intel mac and would like to move the DB over to the new mac. I tried doing the "sudo tar -zcvf snapshot.tgz influxdb " to copy the data over as mentioned in another thread, but now I'm getting multiple bad gateway errors on the new install and not sure where to troubleshoot.  Any help appreciated.

**Comment by jasonacox** [comment]
Hi @rthally ,some things to try or help us help you:

* Did you install the `setup.sh` on the Apple Silicon Mac before the database copy and if so, was it working w/o error?
* Can you paste the errors you are seeing? 
* Try forcing an upgrade `upgrade.sh` 
* Look at the log output from grafana, pypowerwall, and influxdb (e.g. `docker logs influxdb`)
* Bounce all the containers using something like this:
```bash
# stop
./compose-dash.sh stop

# restart
./compose-dash.sh up -d
```

**Comment by rthally** [comment]
> Did you install the setup.sh on the Apple Silicon Mac before the database copy and if so, was it working w/o error?  

Yes, it worked as expected before I did the previously mentioned copy over.

> Can you paste the errors you are seeing?

So I did some fiddling around and I was getting a "permission denied" error in influxdb when I looked at the logs in docker. I did a "brute force" all permissions, and now I get a "InfluxDB: InfluxDB Error: not executed" for the live monitoring and grid status sections. The power meters, analysis and powerwall data areas all say "no data". 

> Try forcing an upgrade upgrade.sh

Already tired. no dice. 

> Look at the log output from grafana, pypowerwall, and influxdb (e.g. docker logs influxdb) 

See previous

> Bounce all the containers using something like this:
> ```
> # stop
> ./compose-dash.sh stop
> # restart
> ./compose-dash.sh up -d
> ```

Tried that multiple times.   Can I just wipe the install and start over? (How?) And is there a better way to get the old data over from my intel mac?

**Reply by jasonacox** [reply]
The issue is with influxdb. I suspect it is an "ownership" issue rather than permission.  Your old mac probably had a different UID (user ID) than your new one.  You will need to update the influxdb directory and files to be owned by the right user:

```bash
# Check for the username that owns the installation
cd Powerwall-Dashboard
ls -ld .
# mine shows: drwxr-xr-x  41 jason  staff  1312 Jul 23 19:58 . 

# Update ownership of influxdb 
docker stop influxdb
sudo chown -R jason:staff influxdb # replace "jason" with your username seen above
docker start influxdb
docker logs influxdb
```

> Tried that multiple times. Can I just wipe the install and start over? (How?) And is there a better way to get the old data over from my intel mac?

Yes, you can wipe it and start over.  However, I'm not aware of a better way to backup and restore the database.  I think you will end up with the same problem with ownership that will require you to chown the files.

**Comment by rthally** [comment]
That ownership issue seemed to fix most of it.  I'm still getting "InfluxDB: InfluxDB Error: not executed" under the energy usage for live monitoring and  grid status.  Stats under power meters appear  and monthly analysis to be good though going back to the beginning.

**Reply by jasonacox** [reply]
Interesting.  I suggest trying:
* Re-install the dashboard.json file
* Check the influxdb logs to see if it gives us more hints `docker logs influxdb`
* Try to "Inspect" one of the panels that is showing the error (and click refresh if nothing shows up)
<img width="341" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/30c9a40e-7889-43c0-b4bc-f297377b2236">

**Comment by rthally** [comment]
Sorry for the dumb question. How do I reinstall the dashboard.json file?Sent from my iPhoneOn Jul 24, 2023, at 22:09, Jason Cox ***@***.***> wrote:﻿
Interesting.  I suggest trying:

Re-install the dashboard.json file
Check the influxdb logs to see if it gives us more hints docker logs influxdb
Try to "Inspect" one of the panels that is showing the error (and click refresh if nothing shows up)

**Reply by jasonacox** [reply]
Not a dumb question... good question.  The instructions here: https://github.com/jasonacox/Powerwall-Dashboard/tree/main#grafana-setup are not very detailed.

<img width="415" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/cb703b05-c348-4519-9723-1e5ac20791d7">

select

<img width="196" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/7e648a31-61aa-41e3-8863-dddcf34d2de4">

pick the dashboard.json from the dashboard folder

<img width="404" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/3b9a1fe8-e130-43a9-b1d7-f6d9db78ea49">

it will warn that you are overwriting - make sure you select the "InfluxDB" and "Sun and Moon" dropdown:

<img width="595" alt="image" src="https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/8f90c7eb-6f87-42d6-8bcf-14a3479eda9f">

**Comment by rthally** [comment]
Sorry for the delay here. I reinstalled the dashboard and no luck.  Here's what the inspector is showing: 

`request:Object
method:"GET"
url:"api/datasources/proxy/1/query"
params:Object
data:null
precision:"ms"
hideFromInspector:false`

`response:Object
results:Array[9]
executedQueryString:"SELECT max(\"home\") FROM \"autogen\".\"http\" WHERE time >= now() - 24h and time <= now() GROUP BY time(2m);SELECT max(\"solar\") FROM \"autogen\".\"http\" WHERE time >= now() - 24h and time <= now() GROUP BY time(2m);SELECT max(from_pw) - max(to_pw) FROM autogen.http WHERE time >= now() - 24h and time <= now() GROUP BY time(2m) ;SELECT max(from_grid) - max(to_grid) FROM autogen.http WHERE time >= now() - 24h and time <= now() GROUP BY time(2m) ;SELECT last(\"percentage\")  /0.95 - 5/0.95 FROM \"autogen\".\"http\" WHERE time >= now() - 24h and time <= now() GROUP BY time(2m) fill(none);SELECT mean(\"temperature\") FROM \"autogen\".\"weather\" WHERE time >= now() - 24h and time <= now() GROUP BY time(2m) fill(null);SELECT mean(\"clouds\") FROM \"autogen\".\"weather\" WHERE time >= now() - 24h and time <= now() GROUP BY time(2m) fill(null);SELECT mean(\"feels_like\") FROM \"autogen\".\"weather\" WHERE time >= now() - 24h and time <= now() GROUP BY time(2m) fill(null);SELECT min(\"grid_status\") +1  FROM \"grid\".\"http\" WHERE grid_status = 0 and time >= now() - 24h and time <= now() GROUP BY time(2m) ;SELECT last(\"backup_reserve_percent\") FROM \"pod\".\"http\" WHERE time >= now() - 24h and time <= now() GROUP BY time(2m) fill(none)"`

**Comment by rthally** [comment]
Any ideas?

**Reply by jasonacox** [reply]
Can you run the verify tool and paste the output?

```bash
./verify.sh 
```

It would also be good to see the output from grafana and influxdb:

```bash
docker logs grafana
docker logs influxdb
```

**Comment by rthally** [comment]
```
Verify Powerwall-Dashboard - Version 2.9.10 - Timezone: America/Los_Angeles
----------------------------------------------------------------------------
This script will attempt to verify all the services needed to run
Powerwall-Dashboard. Use this output when you open an issue for help:
https://github.com/jasonacox/Powerwall-Dashboard/issues/new


Checking pypowerwall
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (pypowerwall): GOOD
 - Service (port 8675): GOOD
 - Version: 0.6.2 Proxy t26
 - Powerwall State: CONNECTED - Firmware Version: 23.12.10

Checking telegraf
----------------------------------------------------------------------------
 - Config File telegraf.conf: GOOD
 - Local Config File telegraf.local: GOOD
 - Container (telegraf): GOOD
 - Version: Telegraf 1.26.1 (git: HEAD@8f421f25)

Checking influxdb
----------------------------------------------------------------------------
 - Config File pypowerwall.env: GOOD
 - Container (influxdb): GOOD
 - Service (port 8086): GOOD
 - Version: InfluxDB shell version: 1.8.10

Checking grafana
----------------------------------------------------------------------------
 - Config File grafana.env: GOOD
 - Container (grafana): GOOD
 - Service (port 9000): GOOD
 - Version: Grafana CLI version 9.1.2

Checking weather411
----------------------------------------------------------------------------
 - Config File weather/weather411.conf: Missing - weather411 not set up
 - Container (weather411): GOOD
 - Service (port 8676): ERROR: Not Listening - Logs:
---
ERROR: No config file. Fix and restart.
Weather411 Server 0.2.1
ERROR: No config file. Fix and restart.
Weather411 Server 0.2.1
ERROR: No config file. Fix and restart.
Weather411 Server 0.2.1
ERROR: No config file. Fix and restart.
Weather411 Server 0.2.1
ERROR: No config file. Fix and restart.
Weather411 Server 0.2.1
ERROR: No config file. Fix and restart.
---
 - Version: Unknown
```

I didn't opt for the weather.  The logs are huge. How many days do you need?

**Reply by jasonacox** [reply]
Thanks!  Just the last screen full (e.g. 50 lines) is fine to start.

**Comment by rthally** [comment]
`ts=2023-08-06T14:10:02.974044Z lvl=info msg="TSM compaction (start)" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 op_event=start
ts=2023-08-06T14:10:02.974319Z lvl=info msg="Beginning compaction" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 tsm1_files_n=4
ts=2023-08-06T14:10:02.974359Z lvl=info msg="Compacting file" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 tsm1_index=0 tsm1_file=/var/lib/influxdb/data/powerwall/monthly/808/000004955-000000002.tsm
ts=2023-08-06T14:10:02.974371Z lvl=info msg="Compacting file" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 tsm1_index=1 tsm1_file=/var/lib/influxdb/data/powerwall/monthly/808/000004963-000000002.tsm
ts=2023-08-06T14:10:02.974421Z lvl=info msg="Compacting file" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 tsm1_index=2 tsm1_file=/var/lib/influxdb/data/powerwall/monthly/808/000004971-000000002.tsm
ts=2023-08-06T14:10:02.974430Z lvl=info msg="Compacting file" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 tsm1_index=3 tsm1_file=/var/lib/influxdb/data/powerwall/monthly/808/000004979-000000002.tsm
ts=2023-08-06T14:10:02.985432Z lvl=info msg="Compacted file" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 tsm1_index=0 tsm1_file=/var/lib/influxdb/data/powerwall/monthly/808/000004979-000000003.tsm.tmp
ts=2023-08-06T14:10:02.985491Z lvl=info msg="Finished compacting files" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 tsm1_files_n=1
ts=2023-08-06T14:10:02.985502Z lvl=info msg="TSM compaction (end)" log_id=0jQq9cZW000 engine=tsm1 tsm1_level=2 tsm1_strategy=level trace_id=0jUnmbNW000 op_name=tsm1_compact_group db_shard_id=808 op_event=end op_elapsed=11.475ms
ts=2023-08-06T14:10:21.093373Z lvl=info msg="Cache snapshot (start)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUnni9G000 op_name=tsm1_cache_snapshot op_event=start
ts=2023-08-06T14:10:21.195605Z lvl=info msg="Snapshot for path written" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUnni9G000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/_internal/monitor/1384 duration=102.262ms
ts=2023-08-06T14:10:21.195660Z lvl=info msg="Cache snapshot (end)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUnni9G000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=102.311ms
ts=2023-08-06T14:17:42.972322Z lvl=info msg="Retention policy deletion check (start)" log_id=0jQq9cZW000 service=retention trace_id=0jUoDgF0000 op_name=retention_delete_check op_event=start
ts=2023-08-06T14:17:42.973640Z lvl=info msg="Retention policy deletion check (end)" log_id=0jQq9cZW000 service=retention trace_id=0jUoDgF0000 op_name=retention_delete_check op_event=end op_elapsed=1.371ms
ts=2023-08-06T14:20:51.093288Z lvl=info msg="Cache snapshot (start)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUoPA5G000 op_name=tsm1_cache_snapshot op_event=start
ts=2023-08-06T14:20:51.184381Z lvl=info msg="Snapshot for path written" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUoPA5G000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/_internal/monitor/1384 duration=91.146ms
ts=2023-08-06T14:20:51.184402Z lvl=info msg="Cache snapshot (end)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUoPA5G000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=91.161ms
ts=2023-08-06T14:31:21.106656Z lvl=info msg="Cache snapshot (start)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUo~c4W000 op_name=tsm1_cache_snapshot op_event=start
ts=2023-08-06T14:31:21.193128Z lvl=info msg="Snapshot for path written" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUo~c4W000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/_internal/monitor/1384 duration=86.566ms
ts=2023-08-06T14:31:21.193153Z lvl=info msg="Cache snapshot (end)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUo~c4W000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=86.578ms
ts=2023-08-06T14:41:51.109641Z lvl=info msg="Cache snapshot (start)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUpb41G000 op_name=tsm1_cache_snapshot op_event=start
ts=2023-08-06T14:41:51.203875Z lvl=info msg="Snapshot for path written" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUpb41G000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/_internal/monitor/1384 duration=94.288ms
ts=2023-08-06T14:41:51.203897Z lvl=info msg="Cache snapshot (end)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUpb41G000 op_name=tsm1_cache_snapshot op_event=end op_elapsed=94.302ms
ts=2023-08-06T14:47:42.986885Z lvl=info msg="Retention policy deletion check (start)" log_id=0jQq9cZW000 service=retention trace_id=0jUpwYYW000 op_name=retention_delete_check op_event=start
ts=2023-08-06T14:47:42.987507Z lvl=info msg="Retention policy deletion check (end)" log_id=0jQq9cZW000 service=retention trace_id=0jUpwYYW000 op_name=retention_delete_check op_event=end op_elapsed=0.669ms
ts=2023-08-06T14:52:21.108777Z lvl=info msg="Cache snapshot (start)" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUqCWy0000 op_name=tsm1_cache_snapshot op_event=start
ts=2023-08-06T14:52:21.188484Z lvl=info msg="Snapshot for path written" log_id=0jQq9cZW000 engine=tsm1 trace_id=0jUqCWy0000 op_name=tsm1_cache_snapshot path=/var/lib/influxdb/data/_internal/monitor/1384 duration=79.771ms
ts=2023-08-06T14:52:21.188510Z lvl=info msg="Cache snapshot (end)" log_id`



`logger=provisioning.alerting t=2023-08-03T12:17:43.148545753Z level=info msg="finished to provision alerting"
logger=ngalert t=2023-08-03T12:17:43.149224003Z level=info msg="warming cache for startup"
logger=provisioning.dashboard t=2023-08-03T12:17:43.149580961Z level=error msg="can't read dashboard provisioning files from directory" path=/var/lib/grafana/provisions/dashboards error="open /var/lib/grafana/provisions/dashboards: no such file or directory"
logger=ngalert.multiorg.alertmanager t=2023-08-03T12:17:43.150670336Z level=info msg="starting MultiOrg Alertmanager"
logger=grafanaStorageLogger t=2023-08-03T12:17:43.150801336Z level=info msg="storage starting"
logger=http.server t=2023-08-03T12:17:43.152850919Z level=info msg="HTTP Server Listen" address=[::]:9000 protocol=http subUrl= socket=
logger=context traceID=00000000000000000000000000000000 userId=1 orgId=1 uname=admin t=2023-08-03T12:17:47.320382255Z level=info msg="Request Completed" method=GET path=/api/live/ws status=0 remote_addr=172.18.0.1 time_ms=3 duration=3.446166ms size=0 referer= traceID=00000000000000000000000000000000
logger=live t=2023-08-03T12:17:47.429185921Z level=info msg="Initialized channel handler" channel=grafana/dashboard/uid/RSabAvRRzZ address=grafana/dashboard/uid/RSabAvRRzZ
logger=context traceID=00000000000000000000000000000000 t=2023-08-03T17:42:47.606432003Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-03T17:42:47.606768128Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=43 duration=43.888458ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 t=2023-08-04T02:42:47.798041917Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-04T02:42:47.798640292Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=86 duration=86.258791ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 t=2023-08-05T07:07:48.217133005Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-05T07:07:48.217534172Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=23 duration=23.740708ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 t=2023-08-06T01:47:48.467472722Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-06T01:47:48.468123139Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=57 duration=57.322125ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 t=2023-08-06T04:47:48.490603429Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-06T04:47:48.491236554Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=34 duration=34.614125ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 t=2023-08-06T06:27:48.525239221Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-06T06:27:48.525567221Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=37 duration=37.592251ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 t=2023-08-06T06:37:48.517358595Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-06T06:37:48.517830887Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=62 duration=62.708708ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000
logger=context traceID=00000000000000000000000000000000 t=2023-08-06T06:37:48.51802022Z level=error msg="Failed to look up user based on cookie" error="user token not found"
logger=context traceID=00000000000000000000000000000000 userId=0 orgId=0 uname= t=2023-08-06T06:37:48.518445054Z level=info msg="Request Completed" method=GET path=/api/datasources/proxy/1/query status=401 remote_addr=172.18.0.1 time_ms=63 duration=63.319041ms size=27 referer="http://localhost:9000/d/RSabAvRRzZ/powerwall-power-flow?from=now-12h&orgId=1&refresh=5m&to=now" traceID=00000000000000000000000000000000`

**Reply by jasonacox** [reply]
Both Grafana and InfluxDB look healthy.  I could still be missing something or it isn't showing up often enough in the logs.

I'm assuming that the "problem" you are still trying to address is that some of the Dashboard panels are still not rendering correctly?  Can you post a screenshot of what you are seeing?   Also, is this only the case after restoring the InfluxdB database?

**Comment by rthally** [comment]
Hi. Screenshot attached.  Correct, everything went crazy after I imported the old db from a different machine to keep my historical data. 

![image](https://github.com/jasonacox/Powerwall-Dashboard/assets/836718/3c801711-f339-44ee-a49c-3eb97b82bdad)

**Reply by jasonacox** [reply]
Thanks @rthally - that helps!  It's almost like there are missing continuous queries (CQ) that feed those two panels.  The meters seem to be working which would indicate the CQ's for power data downsampling are working.

Try to set up the influxDB CQ's again using this:

```bash
  docker exec -it influxdb influx -import -path=/var/lib/influxdb/influxdb.sql
```

Plan B

If the above doesn't work, it seems like something in your old install is not working well with the new one.  I could suggest doing an `upgrade.sh` on the old install to get it to the current version, then doing the backup/restore again.  It could be that your old install is missing several of the newer data elements that the new dashboard.json expects to use.

Plan C

There is an alternative.  You can start over and pull all of your basic "history" from Tesla's cloud using the [tesla-history tool](https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/tesla-history#tesla-history-import-tool) that @mcbirse built. Basically uninstall and reinstall everything.  Something like:

```bash
# remove everything
./compose-dash.sh down
./compose-dash.sh rm
cd ..
sudo rm -fr Powerwall-Dashboard

# reinstall
git clone https://github.com/jasonacox/Powerwall-Dashboard.git
cd Powerwall-Dashboard
./setup.sh

# pull history from Tesla - see https://github.com/jasonacox/Powerwall-Dashboard/tree/main/tools/tesla-history
cd tools/tesla-history
pip install python-dateutil teslapy influxdb
python3 tesla-history.py --login

# get history data for start/end period
python3 tesla-history.py --start "2022-01-01 00:00:00" --end "2023-12-31 23:59:59"
````

**Comment by rthally** [comment]
A) Did nothing that I can see.    B.  Doesn't work because my previous version is so old (more than 18 months), no "upgrade.sh" there.  

Before I do C, is there a way to force the upgrade on B?

**Reply by jasonacox** [reply]
You can force an upgrade:

```bash
# Grab latest upgrade script from github and run it
curl -sL --output tmp.sh https://raw.githubusercontent.com/jasonacox/Powerwall-Dashboard/main/upgrade.sh
bash tmp.sh upgrade
```

**Reply by rthally** [reply]
```
Upgrade Powerwall-Dashboard from Unknown to 2.9.10
---------------------------------------------------------------------
This script will attempt to upgrade you to the latest version without
removing existing data. A backup is still recommended.

ERROR: Missing pypowerwall.env - This means you have not run 'setup.sh' or
       you have an older version that cannot be updated automatically.
       Run 'git pull' and resolve any conflicts then run the 'setup.sh'
       script to re-enter your Powerwall credentials.
```

**Reply by rthally** [reply]
After running git pull: 

```
remote: Enumerating objects: 1064, done.
remote: Counting objects: 100% (581/581), done.
remote: Compressing objects: 100% (169/169), done.
remote: Total 1064 (delta 418), reused 534 (delta 403), pack-reused 483
Receiving objects: 100% (1064/1064), 367.60 KiB | 9.67 MiB/s, done.
Resolving deltas: 100% (632/632), completed with 31 local objects.
From https://github.com/jasonacox/Powerwall-Dashboard
   9114177..1045956  main       -> origin/main
 * [new tag]         v2.8.10    -> v2.8.10
 * [new tag]         v2.8.11    -> v2.8.11
 * [new tag]         v2.8.3     -> v2.8.3
 * [new tag]         v2.8.4     -> v2.8.4
 * [new tag]         v2.8.6     -> v2.8.6
 * [new tag]         v2.8.7     -> v2.8.7
 * [new tag]         v2.8.8     -> v2.8.8
 * [new tag]         v2.8.9     -> v2.8.9
 * [new tag]         v2.9.0     -> v2.9.0
 * [new tag]         v2.9.1     -> v2.9.1
 * [new tag]         v2.9.10    -> v2.9.10
 * [new tag]         v2.9.2     -> v2.9.2
 * [new tag]         v2.9.3     -> v2.9.3
 * [new tag]         v2.9.4     -> v2.9.4
 * [new tag]         v2.9.5     -> v2.9.5
 * [new tag]         v2.9.6     -> v2.9.6
 * [new tag]         v2.9.7     -> v2.9.7
 * [new tag]         v2.9.8     -> v2.9.8
 * [new tag]         v2.9.9     -> v2.9.9
Updating 5d93b46..1045956
error: Your local changes to the following files would be overwritten by merge:
	dashboard.json
	influxdb/.DS_Store
	influxdb/influxdb.sql
	powerwall.yml
	tz
Please commit your changes or stash them before you merge.
error: The following untracked working tree files would be overwritten by merge:
	upgrade.sh
Please move or remove them before you merge.
Aborting
```

**Reply by jasonacox** [reply]
Thanks @rthally !  You will need to stash those changes.  You can do that with this:

```bash
rm upgrade.sh
git stash
git pull --rebase
```

As update helpfully told us, the version is too old to upgrade automatically. So you will need to run setup after that:

```bash
./setup.sh
```
